src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/RcaController.java

```diff
@@ -215,8 +222,13 @@ public void run() {
             checkUpdateNodeRole(nodeDetails);
           }
         }
-
         updateRcaState();
+
+        // Update Analysis graph with Muted RCAs value
+        if (rcaEnabled && rcaScheduler != null && rcaScheduler.getState() == RcaSchedulerState.STATE_STARTED) {
```

Comment:
As we discussed, we want to update the mutedRcas before the call to `updateRcaState()`

Reply:
The mutedRcas is an update to a static datastructure. It should work, whether Rca is enabled or not. But to avoid processing, we can just check for the `rcaEnabled` flag ?

Reply:
We should have a test to make sure that mutedRcas is always accessible

Reply:
> The mutedRcas is an update to a static datastructure. It should work, whether Rca is enabled or not. But to avoid processing, we can just check for the `rcaEnabled` flag ?

Agreed, Have updated to move this under [readRcaEnabledFromConf](https://github.com/opendistro-for-elasticsearch/performance-analyzer-rca/blob/master/src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/RcaController.java#L242)

Reply:
> We should have a test to make sure that mutedRcas is always accessible

Right. I am adding one.